glue/flagging_script_glue/flagging.py

```diff
@@ -767,114 +767,156 @@ def z_normalize_groupby(s: pd.Series):
 
-        labels = [
-            "Home flip sale (high)",
-            "Family sale (high)",
-            "Non-person sale (high)",
-            "Anomaly (high)",
-            "High price swing",
+        # Define labels for price-based reasons
+        price_labels = [
             "High price (raw & sqft)",
-            "High price (raw)",
-            "High price (sqft)",
-            "Home flip sale (low)",
-            "Family sale (low)",
-            "Non-person sale (low)",
-            "Anomaly (low)",
-            "Low price swing",
             "Low price (raw & sqft)",
-            "Low price (raw)",
+            "High price",
+            "Low price",
+            "High price (sqft)",
             "Low price (sqft)",
         ]
 
-    df["sv_outlier_type"] = np.select(conditions, labels, default="Not outlier")
+    # Combined conditions and labels for reason1 with fallback to character conditions if no price condition is met
+    combined_conditions = price_conditions + char_conditions
+    combined_labels = price_labels + char_labels
+
+    df["sv_outlier_reason1"] = np.select(
+        combined_conditions, combined_labels, default="Not outlier"
+    )
+
+    # Adjust for reason2 based on remaining price conditions and char conditions not used in reason1
+    remaining_conditions = [
+        (cond & (df["sv_outlier_reason1"] != label))
+        for cond, label in zip(combined_conditions, combined_labels)
+    ]
+    df["sv_outlier_reason2"] = np.select(
+        remaining_conditions, combined_labels, default="Not outlier"
+    )
+
+    def find_next_unused_char(row, df):
+        used_labels = {row["sv_outlier_reason1"], row["sv_outlier_reason2"]}
+        for cond, label in zip(char_conditions, char_labels):
+            # Evaluate the condition directly for the current row
+            if label not in used_labels and cond[row.name]:
+                return label
+        return "Not outlier"
+
+    # Now we apply this function row-wise correctly by also passing the DataFrame reference:
+    df["sv_outlier_reason3"] = df.apply(find_next_unused_char, df=df, axis=1)
+
+    def transform_row(row):
+        if row["sv_outlier_reason1"] in [
+            "Low price (raw & sqft)",
+            "High price (raw & sqft)",
+        ]:
+            # Determine price type based on the value in sv_outlier_reason1
+            price_type = (
+                "Low price" if "Low" in row["sv_outlier_reason1"] else "High price"
+            )
+
+            # Update sv_outlier_reason1 to just 'Low price' or 'High price'
+            row["sv_outlier_reason1"] = price_type
+
+            # Move the current sv_outlier_reason2 to sv_outlier_reason3
+            row["sv_outlier_reason3"] = row["sv_outlier_reason2"]
+
+            # Update sv_outlier_reason2 to the new value '(sqft)'
+            row["sv_outlier_reason2"] = f"{price_type} (sqft)"
+
+        return row
+
+    # Apply the transformation
+    df = df.apply(transform_row, axis=1)
```

Comment:
This helper function was necessary because the mansueto model outputs "raw and sqft" price outlier information together, and we want to separate them based on the design doc specification. I looked into decoupling this but I think it would have been a massive diff and a lot of work. I think this helper function is the most simple solution.